### PR TITLE
Fix Step Functions LogGroup Error and other database errors

### PR DESCRIPTION
**Pull request by juliusfitzhugh-ccpo**
This PR fixes a few common errors that were seen across the team and implements the changes that fixes those errors (RDS endpoints implementation).
- Add `logGroupName` for the state machine in the stack and use the prefix `/aws/vendedlogs/states/` to over come the `Invalid Logging Configuration: The CloudWatch Logs Resource Policy size was exceeded.` when trying to `cdk deploy`.
- Add entities classes for the connection configuration to prevent `EntityMetadataNotFoundError`.
- Add column types to entities to prevent `CloumnTypeUndefinedError`.

### CODE REVIEW

File: packages/api/utils/database.ts [hunk 1]
```diff
@@ -80,7 +85,7 @@ export async function createConnection(): Promise<Connection> {
     },
     logging: "all",
     database: databaseName,
-    entities: ["/opt/nodejs/orm/entity/**.js"],
+    entities: [Portfolio, Application, Environment, TaskOrder, Clin, "/opt/nodejs/orm/entity/**.js"],
```
Comment (jeffsegal-ccpo) [inline packages/api/utils/database.ts:88]: Will we need to do this forever for all new entities or is this a stop-gap to get through the initial cut of implementations?

**Review by jeffsegal-ccpo — COMMENTED** [review]
Thanks Julius. It's super helpful to collate all these things together in a separate PR so everyone can move together as we get past them.

**Review by jasonburkert-ccpo — APPROVED** [review]
Well done.  Thanks.

File: packages/api/utils/database.ts (hunk 1 above)
Comment (jasonburkert-ccpo) [reply packages/api/utils/database.ts:88]: This is a documented capability.  Reference [this comment](https://github.com/dod-ccpo/atat-web-api/pull/518#discussion_r767985259) where I addressed similar question on #518

Comment (jasonburkert-ccpo) [reply packages/api/utils/database.ts:88]: I think we also need the *.js references when running migrations.

Comment (juliusfitzhugh-ccpo) [reply packages/api/utils/database.ts:88]: I'm not completely sure. @jasonburkert-ccpo found this work around and might have a better idea as if this is needed or if there is a better way.

Comment (jeffsegal-ccpo) [reply packages/api/utils/database.ts:88]: Got it. No big deal for now but we can get to the bottom of it later. Not ideal to have to remember to put new entities in a config file but also not the end of the world.

**Pull request merged**